File: src/main/java/com/bdev/hogwarts_api/data/converter/events/EventModelConverter.java
package com.bdev.hogwarts_api.data.converter.events;

import com.bdev.hogwarts_api.data.dto.events.Event;
import com.bdev.hogwarts_api.data.model.events.EventModel;

import static com.bdev.hogwarts_api.utils.EncodingUtils.fromBase64;

public class EventModelConverter {
    public static Event convert(EventModel eventModel) {
        return Event
                .builder()
                .id(eventModel.getId())
                .eventType(eventModel.getEventType())
                .name(fromBase64(eventModel.getName()))
                .cabinetId(eventModel.getCabinetId())
                .teacherId(eventModel.getTeacherId())
                .date(eventModel.getDate())
                .startTime(eventModel.getStartTime())
                .finishTime(eventModel.getFinishTime())
                .build();
    }
}

File: src/main/java/com/bdev/hogwarts_api/data/converter/events/EventModelConverter.kt
package com.bdev.hogwarts_api.data.converter.events

import com.bdev.hogwarts_api.data.dto.events.Event
import com.bdev.hogwarts_api.data.model.events.EventModel

import com.bdev.hogwarts_api.utils.EncodingUtils.fromBase64

object EventModelConverter {
    fun convert(eventModel: EventModel): Event {
        return Event(
                eventModel.id,
                eventModel.eventType ?: throw RuntimeException(),
                fromBase64(eventModel.name ?: throw RuntimeException()),
                eventModel.cabinetId ?: throw RuntimeException(),
                eventModel.teacherId ?: throw RuntimeException(),
                eventModel.date ?: throw RuntimeException(),
                eventModel.startTime ?: throw RuntimeException(),
                eventModel.finishTime ?: throw RuntimeException()
        )
    }
}
